jaq/src/main.rs

```diff
@@ -217,9 +217,14 @@ fn read_input<'a>(
             Box::new(read.lines().map(|line| Ok(Val::Str(line.unwrap().into()))))
         }
     } else {
-        let inputs = serde_json::Deserializer::from_reader(read)
-            .into_iter::<serde_json::Value>()
-            .map(|r| r.map(Val::from));
+        let inputs = read.lines().map(|line| {
```

Comment:
I was not able to reproduce this PR being faster than serde_json. On my machine it is **slower** than equivalent serde_json code.

I measured using `time target/release/jaq '.text' <yelp_academic_dataset_review.json >/dev/null`.

On this PR rebased to main (2e0e6a695fc15c99b44ad933075c9de45ac70729), it takes 33.8 seconds.

Using equivalent serde_json code it takes 30.1 seconds.

```rust
let inputs = read.lines().map(|line| {
    let line = line.unwrap(); // FIXME
    serde_json::from_str::<serde_json::Value>(&line).map(Val::from)
});
```

So any speedup in the PR is due to the use of `lines`, not the use of `json_deserializer`, with `json_deserializer` actually being slower than `serde_json` for this dataset. As mentioned in https://github.com/01mf02/jaq/pull/50#issuecomment-1307513778, the use of `liens` is not desired by the maintainer.

Reply:
I think you would get significantly faster than both of these (maybe 20-25 seconds?) by `mmap`ing the input file and then using `serde_json::Deserializer::from_slice(input).into_iter()` instead of `from_reader`.

Reply:
A related suggestion: there is probably another several seconds to save by deserializing directly to `Val`, instead of to serde_json::Value and then performing a `From` conversion. This would not be possible with `json_deserializer`.

Reply:
Hi @dtolnay, thanks for your benchmark and your suggestions!
Concerning deserialization to `Val`: I have tried to do so in the past, but could eventually not figure out how to deal with arbitrary-precision numbers, see <https://github.com/serde-rs/json/issues/896>. I would be very happy to find a way to solve this.

Reply:
@dtolnay, following your suggestion, I have now integrated file loading via `mmap` into jaq (971c0dd). I measured performance on a single benchmark, where this change reduces runtime from 6.5 to 4.8 seconds. Thanks a lot!

Reply:
Yet another benchmark run on JSON data from <https://opendata.paris.fr/explore/dataset/denominations-emprises-voies-actuelles/>, runtime decreases from 1208 ms to 880 ms. That is a runtime reduction of about 27% runtime!